File: app/src/main/java/com/example/pam_app/utils/BucketEntryComparator.java
package com.example.pam_app.utils;

import com.example.pam_app.model.BucketEntry;

import java.util.Comparator;

public class BucketEntryComparator implements Comparator<BucketEntry> {

    @Override
    public int compare(final BucketEntry o1, final BucketEntry o2) {
        return o1.getDate().compareTo(o2.getDate());
    }
}

File: app/src/main/java/com/example/pam_app/utils/BucketEntryComparator.kt
package com.example.pam_app.utils

import com.example.pam_app.model.BucketEntry
import java.util.*
import kotlin.jvm.Synchronized

class BucketEntryComparator : Comparator<BucketEntry> {
    override fun compare(o1: BucketEntry, o2: BucketEntry): Int {
        return o1.date?.compareTo(o2.date)!!
    }
}